docs/learning/ECSQLTutorial/ECSqlSyntax.md

```diff
@@ -0,0 +1,1289 @@
+          FROM   [yaxis]
+          WHERE  [y] < 1.0
+        ),
+        [m]([iter], [cx], [cy], [x], [y]) AS(
+          SELECT
+                0,
+                [x],
+                [y],
+                0.0,
+                0.0
+          FROM   [xaxis],
+                [yaxis]
+          UNION ALL
+          SELECT
+                [iter] + 1,
+                [cx],
+                [cy],
+                [x] * [x] - [y] * [y] + [cx],
+                2.0 * [x] * [y] + [cy]
+          FROM   [m]
+          WHERE  ([x] * [x] + [y] * [y]) < 4.0 AND [iter] < 28
+        ),
+        [m2]([iter], [cx], [cy]) AS(
+          SELECT
+                MAX ([iter]),
+                [cx],
+                [cy]
+          FROM   [m]
+          GROUP  BY
+                    [cx],
+                    [cy]
+        ),
+        [a]([t]) AS(
+          SELECT GROUP_CONCAT (SUBSTR (' .+*#', 1 + (CASE WHEN [iter] / 7 > 4 THEN 4 ELSE [iter] / 7 END), 1), '')
+          FROM   [m2]
+          GROUP  BY [cy]
+        )
+      SELECT GROUP_CONCAT (RTRIM ([t]), CHAR (0xa)) mandelbrot_set
+      FROM   [a];
+      /*
+                                        ....#
+                                       ..#*..
+                                     ..+####+.
+                                .......+####....   +
+                               ..##+*##########+.++++
+                              .+.##################+.
+                  .............+###################+.+
+                  ..++..#.....*#####################+.
+                 ...+#######++#######################.
+              ....+*################################.
+     #############################################...
+              ....+*################################.
+                 ...+#######++#######################.
+                  ..++..#.....*#####################+.
+                  .............+###################+.+
+                              .+.##################+.
+                               ..##+*##########+.++++
+                                .......+####....   +
+                                     ..+####+.
+                                       ..#*..
+                                        ....#
+                                        +.
+     */
+```
+
+## GROUP BY clause
+
+Syntax: `GROUP BY <expr-list> [HAVING <group-filter-expr]`
+
+Count instances of each type of class.
+
+```sql
+    SELECT EC_CLASSNAME(ECClassId) ClassName, COUNT(*) InstanceCount
+    FROM [BisCore].[Element]
+    GROUP BY [ECClassId]
+    LIMIT 3
+    /*
+    ClassName                   | InstanceCount
+    ----------------------------|--------------
+    BisCore:DrawingCategory     | 328
+    BisCore:AnnotationTextStyle | 22
+    BisCore:AuxCoordSystem2d    | 2
+    */
+```
+
+Count instances of each type of class by filter out group with count less then 10.
+
+```sql
+    SELECT EC_CLASSNAME(ECClassId) ClassName, COUNT(*) InstanceCount
+    FROM [BisCore].[Element]
```

Comment:
you inconsistently escape stuff ... I think we should make it all stright (BisCore.Element vs [BisCore].[Element]) because it reads better

Reply:
I went through and escaped all queries.